internal/service/ecs/service.go

```diff
@@ -1342,6 +1371,17 @@ func resourceServiceRead(ctx context.Context, d *schema.ResourceData, meta inter
 	if err := d.Set(names.AttrNetworkConfiguration, flattenNetworkConfiguration(service.NetworkConfiguration)); err != nil {
 		return sdkdiag.AppendErrorf(diags, "setting network_configuration: %s", err)
 	}
+
+	if service.Deployments != nil {
```

Comment:
I prefer returning early philosophy, which is Go convention, rather than deeply nested ifs.